docs/data-sources/groups.md

```diff
@@ -30,16 +30,29 @@ data "azuread_groups" "allGroups" {
 }
 ```
 
+*Look up all security groups*
```

Comment:
I think it's worth this subtle distinction due to the aforementioned possibility that M365 groups can act as security groups

```suggestion
*Look up all security-enabled groups*
```